oidcclient/jwk.go

```diff
@@ -20,13 +19,13 @@ import (
 )
 
 const defaultJwkExpiration = 15 * time.Minute
+const zoneIDHeader = "x-zone_uuid"
 
-// OIDCTenant represents one IAS tenant with it's OIDC discovery results and cached JWKs
+// OIDCTenant represents one IAS tenant correlating with one zone with it's OIDC discovery results and cached JWKs
 type OIDCTenant struct {
-	ProviderJSON ProviderJSON
-
-	httpClient   *http.Client
-	singleFlight singleflight.Group
+	ProviderJSON    ProviderJSON
+	acceptedZoneIds map[string]bool
```

Comment:
You can use a `map[string]struct{}` (empty struct) as you don't need to store any value and and empty struct is size 0.
For non accepted or non longer valid zones you can simply `delete()` the key.

Reply:
obsolete

Reply:
We keep boolean, but have to check, whether its ok to keep the zones map for up to 12 hours
(as cache entry gets kept up to 12h - [see here](https://github.com/SAP/cloud-security-client-go/blob/master/auth/middleware.go#L24) to avoid token-keys endpoint discovery)